File: TripKitAndroid/src/main/java/com/skedgo/tripkit/ExternalActionParams.java
package com.skedgo.tripkit;

import androidx.annotation.Nullable;

import com.skedgo.tripkit.routing.TripSegment;

import org.immutables.value.Value;

import static org.immutables.value.Value.Style.BuilderVisibility.PACKAGE;
import static org.immutables.value.Value.Style.ImplementationVisibility.PRIVATE;

@Value.Immutable
@Value.Style(visibility = PRIVATE, builderVisibility = PACKAGE)
public abstract class ExternalActionParams {
  public static Builder builder() {
    return new ExternalActionParamsBuilder();
  }

  public abstract String action();
  public abstract TripSegment segment();
  @Nullable public abstract String flitWaysPartnerKey();

  public interface Builder {
    Builder action(String action);
    Builder segment(TripSegment segment);
    Builder flitWaysPartnerKey(String flitWaysPartnerKey);
    ExternalActionParams build();
  }
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/ExternalActionParams.kt
package com.skedgo.tripkit

import com.skedgo.tripkit.routing.TripSegment
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style
import org.immutables.value.Value.Style.BuilderVisibility.PACKAGE
import org.immutables.value.Value.Style.ImplementationVisibility.PRIVATE

@Immutable
@Style(visibility = PRIVATE, builderVisibility = PACKAGE)
abstract class ExternalActionParams {
    abstract fun action(): String

    abstract fun segment(): TripSegment

    abstract fun flitWaysPartnerKey(): String?

    interface Builder {
        fun action(action: String): Builder

        fun segment(segment: TripSegment?): Builder

        fun flitWaysPartnerKey(flitWaysPartnerKey: String): Builder

        fun build(): ExternalActionParams
    }

    companion object {
        fun builder(): Builder {
            return ExternalActionParamsBuilder()
        }
    }
}